include/RAJA/policy/sycl/teams.hpp

```diff
@@ -0,0 +1,1203 @@
+/*!
+ ******************************************************************************
+ *
+ * \file
+ *
+ * \brief   RAJA header file containing user interface for RAJA::Teams::sycl
+ *
+ ******************************************************************************
+ */
+
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-22, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef RAJA_pattern_teams_sycl_HPP
+#define RAJA_pattern_teams_sycl_HPP
+
+#include "RAJA/pattern/teams/teams_core.hpp"
+#include "RAJA/pattern/detail/privatizer.hpp"
+#include "RAJA/policy/sycl/policy.hpp"
+#include "RAJA/policy/sycl/MemUtils_SYCL.hpp"
+//#include "RAJA/policy/sycl/raja_syclerrchk.hpp"
+#include "RAJA/util/resource.hpp"
+
+namespace RAJA
+{
+
+namespace expt
+{
+
+template <bool async> //switch 1 -> 0, but what should it be ? Ask R. Chen?
+struct LaunchExecute<RAJA::expt::sycl_launch_t<async, 0>> {
```

Comment:
@rchen20 do you have any thoughts on what the value to the right of async should be? It was switched to 1 for cuda, but not sure what it should be here.

Reply:
I had purposely not implemented a `sycl_launch` policy because I'm not sure what the equivalent of "blocks per SM" would be in SYCL, or if SYCL can support a blocks per SM input upon launch. Perhaps @homerdin would know?

Reply:
What would you do in sycl? The policies express what is available in the backend (sycl) not necessarily what's available in the HW or implementation.

Reply:
What do we need that for? SYCL doesn't expose that directly AFAIK, but you can query all the necessary properties to compute it. At least you can through the underlying opencl, these are the two I would look to for this (CL_​DEVICE_​MAX_​COMPUTE_​UNITS probably most relevant, combined with kernel info for how many can actually be resident):

* https://www.khronos.org/registry/OpenCL/sdk/2.2/docs/man/html/clGetDeviceInfo.html
* https://www.khronos.org/registry/OpenCL/sdk/2.2/docs/man/html/clGetKernelWorkGroupInfo.html